### PR TITLE
*: Remove useless code and use the DAG mode in tests

### PR DESCRIPTION
**Pull request by zimulala**
Clean up the package of `mocktikv`.

### CODE REVIEW

**Comment by hanfei1991** [comment]
LGTM

**Comment by hanfei1991** [comment]
https://github.com/pingcap/tidb/pull/3069/files#diff-9f657f87f877075313b8774ebe81ae86R593
this can be removed

**Comment by hanfei1991** [comment]
https://github.com/pingcap/tidb/blob/master/store/tikv/mock-tikv/executor.go#L593
can be removed

**Comment by hanfei1991** [comment]
https://github.com/pingcap/tidb/blob/master/store/tikv/mock-tikv/executor.go#L658
can be moved to cop_handle_dag.go

**Comment by hanfei1991** [comment]
cop_handle_dag.go -> cop_handle.go ?

File: store/tikv/mock-tikv/cop_handler_dag.go [hunk 1]
```diff
@@ -292,6 +294,105 @@ func (e *evalContext) decodeRelatedColumnVals(relatedColOffsets []int, handle in
...
+	if len(chunks) == 0 || len(chunks[len(chunks)-1].RowsMeta) >= rowsPerChunk {
+		chunks = append(chunks, tipb.Chunk{})
+	}
+	cur := &chunks[len(chunks)-1]
+	cur.RowsMeta = append(cur.RowsMeta, tipb.RowMeta{Handle: handle, Length: int64(len(data))})
+	cur.RowsData = append(cur.RowsData, data...)
+	return chunks
+}
+
+func maxStartKey(rangeStartKey kv.Key, regionStartKey []byte) []byte {
```
Comment (hanfei1991) [inline store/tikv/mock-tikv/cop_handler_dag.go]: move to executor.go

File: store/tikv/mock-tikv/cop_handler_dag.go [hunk 2]
```diff
@@ -292,6 +294,105 @@ func (e *evalContext) decodeRelatedColumnVals(relatedColOffsets []int, handle in
...
+	if bytes.Compare([]byte(rangeStartKey), regionStartKey) > 0 {
+		return []byte(rangeStartKey)
+	}
+	return regionStartKey
+}
+
+func minEndKey(rangeEndKey kv.Key, regionEndKey []byte) []byte {
```
Comment (hanfei1991) [inline store/tikv/mock-tikv/cop_handler_dag.go]: ditto

File: store/tikv/mock-tikv/cop_handler_dag.go [hunk 3]
```diff
@@ -292,6 +294,105 @@ func (e *evalContext) decodeRelatedColumnVals(relatedColOffsets []int, handle in
...
+	if len(regionEndKey) == 0 || bytes.Compare([]byte(rangeEndKey), regionEndKey) < 0 {
+		return []byte(rangeEndKey)
+	}
+	return regionEndKey
+}
+
+func decodeHandle(data []byte) (int64, error) {
```
Comment (hanfei1991) [inline store/tikv/mock-tikv/cop_handler_dag.go]: ditto

File: store/tikv/mock-tikv/cop_handler_dag.go (hunk 1 above)
Comment (zimulala) [inline store/tikv/mock-tikv/cop_handler_dag.go]: Why? This file also uses this function.

File: store/tikv/mock-tikv/cop_handler_dag.go (hunk 2 above)
Comment (zimulala) [inline store/tikv/mock-tikv/cop_handler_dag.go]: Ditto.

File: store/tikv/mock-tikv/cop_handler_dag.go (hunk 1 above)
Comment (hanfei1991) [inline store/tikv/mock-tikv/cop_handler_dag.go]: ok, I'm blind

File: store/tikv/mock-tikv/cop_handler_dag.go (hunk 2 above)
Comment (hanfei1991) [inline store/tikv/mock-tikv/cop_handler_dag.go]: blind two times

**Comment by hanfei1991** [comment]
LGTM

**Pull request merged**